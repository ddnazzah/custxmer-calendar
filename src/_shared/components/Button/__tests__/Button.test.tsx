/*
 * @Author: Dieu-Donne Nazzah
 * @Date: 2021-06-28 04:19:16
 * @Last Modified by: Dieu-Donne Nazzah
 * @Last Modified time: 2021-12-01 05:45:56
 */

import { fireEvent, render } from '@testing-library/react-native';
import React from 'react';
import { Button } from '../';
import TestProvider from '../../../../@testing/Provider';

describe('<Button/>', () => {
    it('renders correctly', () => {
        const { toJSON } = render(
            <TestProvider>
                <Button />
            </TestProvider>,
        );

        expect(toJSON()).toMatchSnapshot();
    });

    it('renders text correctly', () => {
        const { getByText } = render(
            <TestProvider>
                <Button text="My Button" />
            </TestProvider>,
        );

        expect(getByText('My Button')).toBeDefined();
    });

    it('calls onPress', () => {
        const onPress = jest.fn();

        const { getByTestId } = render(
            <TestProvider>
                <Button testID="button" text="Button" onPress={onPress} />
            </TestProvider>,
        );

        const button = getByTestId('button');

        fireEvent.press(button);

        expect(onPress).toHaveBeenCalledTimes(1);
    });

    it('disables', () => {
        const onPress = jest.fn();

        const { getByTestId } = render(
            <TestProvider>
                <Button
                    testID="button"
                    text="Button"
                    onPress={onPress}
                    disabled
                />
            </TestProvider>,
        );

        const button = getByTestId('button');

        fireEvent.press(button);

        expect(onPress).toHaveBeenCalledTimes(0);
    });
});
