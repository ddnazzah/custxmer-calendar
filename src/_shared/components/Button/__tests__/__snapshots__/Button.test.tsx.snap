// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> renders correctly 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F75843",
      "borderRadius": 5,
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 15,
      "paddingVertical": 12,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#ffffff",
          "fontSize": 15,
        },
      ]
    }
  />
</View>
`;
